bootstrap/src/main/scala/org/scalafmt/bootstrap/Bootstrap.scala

```diff
@@ -41,11 +41,14 @@ object ScalafmtBootstrap {
   private val cliCache =
     mutable.Map.empty[String, Either[Throwable, ScalafmtBootstrap]]
 
-  def fromVersion(version: String): Either[Throwable, ScalafmtBootstrap] =
-    cliCache.getOrElseUpdate(version, fromVersionUncached(version))
+  def fromVersion(version: String): Either[Throwable, ScalafmtBootstrap] = {
+    synchronized {
```

Comment:
Yea I think so.

Otherwise a second thread could pass these lines and find the cache empty while a first thread is busy running fromVersionUncached. Once the first has finished, the second will perform fromVersionUncached without knowing that its just been put in the cache.

So here its more of a performance optimisation than avoiding the coursier locking.